README.md

```diff
@@ -157,7 +170,45 @@ MatomoTracker.instance.trackScreenWithName(
     );
 ```
 
-## Contributors
+# Migration Guide
+
+## v3.0.0
+
+Now the `initialize()` method takes a `LocalStorage? localStorage` instead of a `SharedPreferences? prefs` as its parameter to override the persistent data implementation.
+
+By default it will use an implementation of [shared_preferences](https://pub.dev/packages/shared_preferences) with the class `SharedPrefsStorage`, but you can provide your own implementation of `LocalStorage` to use a different package.
+
+### Old
+
+```dart
+final myPrefs = await SharedPreferences.getInstance();
+
+await MatomoTracker.instance.initialize(
+    siteId: siteId,
+    url: 'https://example.com/matomo.php',
+    prefs: myPrefs,
+);
+```
+
+### New
+
+```dart
+class MyLocalStorage implements LocalStorage {
+    MyLocalStorage();
+
+    // ...
+}
+
+final myStorage = MyLocalStorage();
+
+await MatomoTracker.instance.initialize(
```

Comment:
We should demonstrate to the user that we can also pass no input. I believe that if people read quickly, they will not get this.